src/librustc/ty/print/pretty.rs

```diff
@@ -920,28 +920,46 @@ pub trait PrettyPrinter<'tcx>:
         Ok(self)
     }
 
-    fn pretty_print_const_value(
+    fn pretty_print_const_scalar(
         mut self,
-        ct: ConstValue<'tcx>,
+        scalar: Scalar,
         ty: Ty<'tcx>,
     ) -> Result<Self::Const, Self::Error> {
         define_scoped_cx!(self);
 
-        if self.tcx().sess.verbose() {
-            p!(write("ConstValue({:?}: {:?})", ct, ty));
-            return Ok(self);
-        }
-
         let u8 = self.tcx().types.u8;
 
-        match (ct, &ty.kind) {
-            (ConstValue::Scalar(Scalar::Raw { data, .. }), ty::Bool) =>
-                p!(write("{}", if data == 0 { "false" } else { "true" })),
-            (ConstValue::Scalar(Scalar::Raw { data, .. }), ty::Float(ast::FloatTy::F32)) =>
+        match (scalar, &ty.kind) {
+            // Single element arrays print their element (they are `#[transparent]`) enclosed in
+            // square brackets.
+            (_, ty::Array(t, n)) if n.eval_usize(self.tcx(), ty::ParamEnv::empty()) == 1 => {
+                p!(write("["));
+                self = self.pretty_print_const_scalar(scalar, t)?;
+                p!(write("]"));
+            },
+            // Byte strings (&[u8; N])
+            (
+                Scalar::Ptr(ptr),
+                ty::Ref(_, ty::TyS { kind: ty::Array(t, n), .. }, _),
+            ) if *t == u8 => {
+                let n = n.eval_usize(self.tcx(), ty::ParamEnv::empty());
+                let byte_str = self.tcx()
+                    .alloc_map.lock()
+                    .unwrap_memory(ptr.alloc_id)
+                    .get_bytes(&self.tcx(), ptr, Size::from_bytes(n)).unwrap();
+                p!(pretty_print_byte_str(byte_str));
+            },
+            // Bool
+            (Scalar::Raw { data: 0, .. }, ty::Bool) => p!(write("false")),
+            (Scalar::Raw { data: 1, .. }, ty::Bool) => p!(write("true")),
+            (Scalar::Raw { data, .. }, ty::Bool) => p!(write("{{invalid bool: {:02x}}}", data)),
```

Comment:
Couldn't this print like `u8` except show `bool` instead of `u8`?
But maybe when we get full ADT printing we'd have an uniform "invalid" state (e.g. `enum` discriminants).

Reply:
you mean just `{0x42 : bool}` ?

Reply:
The former, I guess that's not how you print bools though.
I don't think you need any change here anyway.